### ISSUE TITLE
[PROPOSAL] Refactor DatePicker to be more accessible to screen reader and keyboard users

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#4763]
~~The current implementation of our DatePicker has serious usability and keyboard accessibility issues. We've been relying on a third party library for our calendar drop down functionality and a refactor involving an in-house solution could greatly simplify component API, debugging, and accessibility bug fixes moving forward.~~

~~As for guidance -- [there are browser date picker defaults](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date) that are quite accessible and building on top or these seems like the smart move. As an alternative the[ jQuery UI Datepicker](https://jqueryui.com/datepicker/) has historically been highly regarded from a keyboard accessibility/screen reader standpoint.~~

**Re-evaluating our Datepicker's calendar dropdown functionality has value as well. I suggest we take a cue from [the Gov.uk design system](https://design-system.service.gov.uk/components/date-input/) and rethink this interaction altogether.** 

_After talking about this last year (circa October?) we chose to shore up any existing issues and move forward with FlatPicker as our calendar libarary_ 

### Update 2/10/20
ref #5310 

Given this latest crop of accessibility issues for keyboard, JAWS, and VO -- I think there's value in returning to the discussion of alternatives here if the cost makes sense. For example [Airbnb's React-Dates](https://airbnb.io/react-dates) seems very thoughtfully implemented in terms of screen reader a11y and reads quite well in JAWS and VoiceOver

@carbon-design-system/developers pinging the team for thoughts and input

**Comment by joshblack** [comment]
@dakahn for rethinking the interaction, would this be something where we want to include design?

**Comment by dakahn** [comment]
reopening as per request by @snidersd 

We have a new batch of keyboard and screen reader issues with our DatePicker w/Calendar :hurtrealbad:

**Comment by dakahn** [comment]
Closing this up again. Conversation was DOA and it looks like we're just going to continue with flatpickr for the time being
